float32 q0
float32 q1
float32 q2
float32 q3
